lib/tests/test_rewrite.rs

```diff
@@ -1612,3 +1612,76 @@ fn test_empty_commit_option(empty_behavior: EmptyBehaviour) {
         }
     );
 }
+
+#[test]
+fn test_rebase_commit_via_rebase_api() {
+    let settings = testutils::user_settings();
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+
+    // Rebase B onto B2
+    //
+    // D (empty)  E (WC, empty)
+    // |         /
+    // C---------
+    // |
+    // B B2
+    // |/
+    // A
```

Comment:
I think we should explain what the result should be and why. Something like:

> Commit B should be abandoned since it becomes empty after the rebase.
> Commit D should be abandoned since it was empty even before the rebase.
> Commit E should *not* be abandoned, because it is the working copy.
> So, the result should be:
>
>   E' (WC, empty)
>   |
>   C'
>   |
>   B2
>   |
>   A